examples/dataflow-custom-templates/README.md

```diff
@@ -0,0 +1,273 @@
+<!--
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Overview
+
+The purpose of this walkthrough is to create
+[Custom Dataflow templates](https://cloud.google.com/dataflow/docs/concepts/dataflow-templates).
+
+The value of Custom Dataflow templates is that it allows us to execute
+Dataflow jobs without installing any code.  This is useful to enable Dataflow
+execution using an automated process or to enable others without technical
+expertise to run jobs via a user-friendly guided user interface.
+
+You have two options to walkthrough the steps to deploy this example:
+
+1. See [Easy Walkthrough](README.md#Easy%20Walkthrough) to walk through the steps
+without installing anything on your local machine.
+2. See [Requirements](README.md#Requirements) and beyond for an unguided approach.
+
+# Easy Walkthrough 🏖️
+
+For an easy walkthrough without installing anything on your local machine:
+
+[![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.svg)](https://shell.cloud.google.com/cloudshell/editor?cloudshell_git_repo=https%3A%2F%2Fgithub.com%2FGoogleCloudPlatform%2Fprofessional-services&cloudshell_git_branch=main&cloudshell_workspace=examples%2Fdataflow-custom-templates&cloudshell_tutorial=cloud_shell_tutorial.md)
```

Comment:
If you would like to test this walkthrough for this PR, in the URL substitute:

1. `GoogleCloudPlatform` to `damondouglas`
2. `main` to `dataflow-custom-templates`